runelite-client/src/main/java/net/runelite/client/plugins/tearsofguthix/TearsOfGuthixOverlay.java

```diff
@@ -87,6 +89,15 @@ else if (object.getId() == ObjectID.GREEN_TEARS ||
 				? (double) duration.toMillis() / MAX_TIME.toMillis()
 				: 1));
 
+			if (!config.showGreenTearsTimer())
+			{
+				if (object.getId() == ObjectID.GREEN_TEARS ||
+					object.getId() == ObjectID.GREEN_TEARS_6666)
+				{
+					return;
+				}
+			}

```

Comment:
Personally, I'd put this check either at the beginning of the lambda or at L68 just after the `if (position == null)`. Would avoid calculating and prepping the pie component when it's not going to be used anyway.

Reply:
This was a choice I made while during development, noticing that if I toggled the feature, the green timer would be incorrect until the stream moved.

I'm not sure that the calculations are at such a cost that it would be worth introducing this behaviour.

Reply:
You must be thinking of having done that on the plugin side such that it would affect what `plugin.getStreams()` returns.

Moving the check on the overlay side up a few lines will not introduce that behaviour.
I ran that change on my end to confirm it doesn't cause any issues.

Reply:
Oops. Not used to GitHub here, intended to comment while resolving. Intended message:

Indeed it is! Sorry about that, I had some difficulties getting the build running with tests so I drew from memory at the time of commenting.

Looks good on my end, I've pushed up an amended commit addressing this change.

Reply:
You can move it above even the `getCanvasLocation` call. 

Also do:

```java
			if ((object.getId() == ObjectID.GREEN_TEARS || object.getId() == ObjectID.GREEN_TEARS_6666)
				&& !config.showGreenTearsTimer())
			{
				return;
			}
```

Since the id check is faster than the config check.

Reply:
Good call, I'll make this change now. Thanks, @Adam- !